AlphaWallet/WalletConnect/Providers/v2/WalletConnectV2Client.swift

```diff
@@ -19,6 +20,37 @@ struct SocketFactory: WebSocketFactory {
     }
 }
 
+struct DefaultEthereumSignerFactory: SignerFactory {
+    func createEthereumSigner() -> WalletConnectSwiftV2.EthereumSigner {
+        Web3Signer()
+    }
+}
+
+struct Web3Signer: WalletConnectSwiftV2.EthereumSigner {
+    enum SignerError: Error {
+        case recoverPubKeyFailure
+    }
+
+    func sign(message: Data, with key: Data) throws -> EthereumSignature {
```

Comment:
nope, not sure how to call it

Reply:
this method get called when signing `auth message`, while receiving auth request. As i understood its our private key with what we need to pass when receiving request

Reply:
`AuthRequestInteractor` file from example project

Reply:
@hboon do i understand right, we need our own version for signing [AuthPayload](doc://com.apple.documentation.unknown/documentation/auth/authpayload) payload, right?

Reply:
Can you point me to the example project/AuthRequestInteractor where it's used? Not sure how it's supposed to work. I think since the sign function is in a protocol the SDK provides and the function includes a private key argument, the comes from them. If it's ours, then we just wouldn't call this :)

Reply:
https://github.com/WalletConnect/WalletConnectSwiftV2/blob/53e3c5f290a29a88cf98bd7c9d1da1be031a3012/Example/WalletApp/PresentationLayer/Wallet/AuthRequest/AuthRequestPresenter.swift#L30

Reply:
AuthRequestPresenter view appears when we receive auth request, on approve action we get `onApprove()` event

Reply:
for signing `let signature = try signer.sign(` there is private key defined in example app, so i am not sure that to do next, make own copy for cacao signature or ..

Reply:
Hmm.. Ok, let's leave it first if we have no idea. I will try to look at it again. It guess it's harmless to merge as long as we don't call it ourselves (since key always have to come from the caller)

Reply:
ok

Reply:
@oa-s Let's continue (and merge), but can you help ask them separate about this call and who calls it and when, passing in the private key? Since we aren't calling it ourselves, it's ok, but should help us to learn more about how it works